static/js/stream_edit.js

```diff
@@ -461,6 +461,7 @@ export function archive_stream(stream_id, $alert_element, $stream_row) {
         },
         success() {
             $stream_row.remove();
+            stream_settings_ui.redraw_left_panel();
```

Comment:
I think we can avoid calling `redraw_left_panel` on archiving the stream. As that will re-create the complete panel. We can just check if there are any streams to show and to show the text or not accordingly.

Reply:
if we want to check that, we have to call major functions that redraw_left_panel() is calling because of hidden ids concept, that's why I called itself.
Should we remove this ?

Reply:
And after creating a stream, if we do same thing than there would be a glitch because of stream added firstly than removing the message.
The best solution for this is to redraw the left panel.

Reply:
Can't we just check using jquery whether any stream row exists or not by using `length` attribute for an appropriate selector?